File: models/pharmacy/MedicineMaster.js
import mongoose from "mongoose";
// ye commong hoga agar meri 10 branch hai toh ye toh comon hi hoga because ye ek  medicie ka collection hai jo har branch me use hoga so that har branch ke liye alag alag create na karni pade 
// Medicine Master (Drug Catalog)
const MedicineMasterSchema = new mongoose.Schema({

  hospitalId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Hospital",
    required: true
  },

  name: { type: String, required: true },
  genericName: String,
  brandName: String,

  category: String,

  scheduleType: {
    type: String,
    enum: ["OTC","ScheduleH","ScheduleX"]
  },

  form: String,
  strength: String,

  manufacturer: String,

  hsnCode: String,
  gstPercentage: Number,

  isActive: { type: Boolean, default: true },

  createdBy: { type: mongoose.Schema.Types.ObjectId, ref: "User" }

}, { timestamps: true });

export default mongoose.model("MedicineMaster", MedicineMasterSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/MedicineMaster.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "medicine_masters")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MedicineMaster extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @Column(nullable = false)
    private String name;

    @Column(name = "generic_name")
    private String genericName;

    @Column(name = "brand_name")
    private String brandName;

    private String category;

    @Enumerated(EnumType.STRING)
    @Column(name = "schedule_type")
    private ScheduleType scheduleType;

    private String form;
    private String strength;

    private String manufacturer;

    @Column(name = "hsn_code")
    private String hsnCode;

    @Column(name = "gst_percentage")
    private Double gstPercentage;

    @Column(name = "is_active")
    private boolean isActive = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    public enum ScheduleType {
        OTC, ScheduleH, ScheduleX
    }
}
